commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/UsernameFilter.java

```diff
@@ -0,0 +1,104 @@
+package org.wildfly.extras.creaper.commands.security.realms;
+
+/**
+ * Helper class which holds information username-filter username to dn in ldap authorization of security realm.
+ */
+public final class UsernameFilter {
+
+    private final String attribute;
+    private final String baseDn;
+    private final Boolean force;
+    private final Boolean recursive;
+    private final String userDnAttribute;
+    private final LdapCache cache;
+
+    private UsernameFilter(Builder builder) {
+        this.baseDn = builder.baseDn;
+        this.attribute = builder.attribute;
+        this.recursive = builder.recursive;
+        this.userDnAttribute = builder.userDnAttribute;
+        this.force = builder.force;
+        this.cache = builder.cache;
+    }
+
+    public String getAttribute() {
+        return attribute;
+    }
+
+    public String getBaseDn() {
+        return baseDn;
+    }
+
+    public Boolean getForce() {
+        return force;
+    }
+
+    public Boolean getRecursive() {
+        return recursive;
+    }
+
+    public String getUserDnAttribute() {
+        return userDnAttribute;
+    }
+
+    public LdapCache getCache() {
+        return cache;
+    }
```

Comment:
Do the methods have to be `public`? I mean, is the class meant to be used for other things than creating a command? If it's only for creating the command, the methods can be omitted and the `private` variables made package-protected.